codec/src/main/java/com/nixxcode/jvmbrotli/dec/Decoder.java

```diff
@@ -0,0 +1,63 @@
+/* Copyright 2017 Google Inc. All Rights Reserved.
+
+   Distributed under MIT license.
+   See file LICENSE for detail or copy at https://opensource.org/licenses/MIT
+*/
+
+package com.nixxcode.jvmbrotli.dec;
```

Comment:
We don't want these files copy-pasted into our tree. Use reflection+method handles to find them and dispatch at runtime.

Reply:
Mhhmmmmmmm.... I will be honest. I never did reflections and method handles. So need help.

Reply:
Still a newbie. :(

Reply:
There is a code change in the Decoder of JVMBrotli. How I will tackle it when using Reflections?

Reply:
You can use `Class.forName()` to test if a class is present on the class path. Method handles are not available until Java 8, I think, so you need to guard that code with Java version checks. The `Conscrypt` class shows an example of how this is done.
If you can't use `Decoder` directly, you can you the `BrotliDecoderChannel` instead, right?

Reply:
Original Decoder doesn't have `NEEDS_MORE_INPUT` case. I need that to determine whether I need to keep aggregating. I don't think it's possible to modify source code on Runtime or something.

We need some other way to tackle this.

Reply:
A block can be up to 16 MB. So if a chunk is part of the block, we need to aggregate it. Once we have all chunks of a block, we will pass to the decoder and decompress the block and pass it down to the pipeline.

Reply:
@hyperxpro if the brotli implementation not contains all we need you may want to consider open a PR against it... We can revisit once all needed features are present.

Reply:
@normanmaurer Official Brotli for Java is not updated on Maven for a while (only being developed on GitHub). So even if we get the feature after doing PR against it, we can't use it directly. JVMBrotli is also not updated for a year so it's inactive too.

So I think to fork the project, make changes, and publish my modified forked version on Maven with the functionality. How does it sound?

Reply:
@normanmaurer How do you compile binaries for multiple platforms like Windows, Linux, and macOS? Using VM or something?